samplers/src/test/java/internal/CelBasedSamplerComponentProviderTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package internal;
+
+import static io.opentelemetry.sdk.trace.samplers.SamplingResult.recordAndSample;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import dev.cel.common.CelValidationException;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigException;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.contrib.sampler.CelBasedSampler;
+import io.opentelemetry.contrib.sampler.CelBasedSamplerBuilder;
+import io.opentelemetry.contrib.sampler.internal.CelBasedSamplerComponentProvider;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfiguration;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import io.opentelemetry.semconv.HttpAttributes;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class CelBasedSamplerComponentProviderTest {
+
+  private static final CelBasedSamplerComponentProvider PROVIDER =
+      new CelBasedSamplerComponentProvider();
+
+  @Test
+  void endToEnd() {
+    String yaml =
```

Comment:
This is _just_ long enough that I think maybe a resource file could be easier to maintain, but definitely in this first pass this is fine.

Reply:
All the config parts are now move to test resources and are loaded from disk relative to the class.